### ISSUE TITLE
Can I really use glob in entitiesDirs?

### ISSUE THREAD

**Issue by jeiea** [mikro-orm/mikro-orm#341]
**Describe the bug**
[Latest doc](https://github.com/mikro-orm/mikro-orm/blob/master/docs/docs/installation.md) says `entitiesDirs: ['./dist/app/**/entities']` should work, but it seems not.

**To Reproduce**
```
Welcome to Node.js v13.5.0.       
Type ".help" for more information.
> var mikro = require('mikro-orm');
undefined
> var inst = mikro.MikroORM.init({
...   clientUrl: 'db:5432',
...   dbName: 'database',
...   entitiesDirs: ['./dist/**/entity'],
...   entitiesDirsTs: ['./src/**/entity'],
...   type: 'postgresql',
... });
undefined
> (node:11304) UnhandledPromiseRejectionWarning: ValidationError: No entities were discovered
    at Function.noEntityDiscovered (project\node_modules\mikro-orm\dist\utils\ValidationError.js:85:16)
    at MetadataValidator.validateDiscovered (project\node_modules\mikro-orm\dist\metadata\MetadataValidator.js:30:43)
    at MetadataDiscovery.findEntities (project\node_modules\mikro-orm\dist\metadata\MetadataDiscovery.js:61:24)      
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async MetadataDiscovery.discover (project\node_modules\mikro-orm\dist\metadata\MetadataDiscovery.js:30:9)     
    at async Function.init (project\node_modules\mikro-orm\dist\MikroORM.js:45:24)
(node:11304) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)        
(node:11304) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

**Expected behavior**
One entity is found as described below.

**Additional context**
```
> var normalize = require('path').normalize
undefined
> var gl = require('globby');
undefined
>
> function normalizePath(...parts) {
...   let path = parts.join('/').replace(/\\/g, '/').replace(/\/$/, '');
...   path = normalize(path).replace(/\\/g, '/');
...
...   return path.match(/^[/.]|[a-zA-Z]:/) ? path : './' + path;
... }
undefined
> // Maybe current mikro-orm discovery logic
> gl.sync('*', { cwd: normalizePath('.', './dist/**/entity')})
[]
> // What I expected
> gl.sync('./dist/**/entity/*')
[ './dist/entity/HitPost.js' ]
```

**Versions**

| Dependency | Version |
| - | - |
| node | 13.5.0 |
| typescript | 3.7.5 |
| mikro-orm | 3.1.1 |
| your-driver | pg 7.18.1 |

**Comment by B4nan** [comment]
First of all, enable debug mode so you can see if it finds some files inside the directory. `No entities were discovered` may also mean that the file was found but it was not considered as an entity. Also please provide that entity definition. 

Another suggestion, check what path is currently set as cwd (`console.log(process.cwd())`), if that does not reflect your current directory, use `baseDir` option to fix it. 

Have you also tried `entitiesDirs: ['./dist/entity'],`?

**Comment by jeiea** [comment]
Yeah, I already tried your suggestion, and that worked. Then isn't it
globbing problem?

2020년 2월 19일 (수) 오전 5:45, Martin Adámek <notifications@github.com>님이 작성:

**Comment by B4nan** [comment]
globby is used here: 

https://github.com/mikro-orm/mikro-orm/blob/master/lib/metadata/MetadataDiscovery.ts#L74

you could try debugging what value is actually there. it might be also some windows related problem...

**Comment by B4nan** [comment]
Fixed in 3.2
